ethcore/src/trace/filter.rs

```diff
@@ -0,0 +1,255 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::ops::Range;
+use bloomchain::{Filter as BloomFilter, Bloom, Number};
+use util::{Address, FixedHash};
+use util::sha3::Hashable;
+use basic_types::LogBloom;
+use super::flat::FlatTrace;
+use super::trace::Action;
+
+/// Traces filter.
+pub struct Filter {
+	/// Block range.
+	pub range: Range<usize>,
+
+	/// From address. If empty, match all, if not, match one of the values.
```

Comment:
If I use enum type eg.

```rust
pub enum FilterAddresses {
  All
  Some(HashSet<Address>)
}
```

I would have to check if `Some` is empty... But having `pub struct FilterAddresses(HashSet<Address>)` sounds good :)